resources/api/openapi.yml

```diff
@@ -1,454 +1,696 @@
+      properties:
+        customer:
+          $ref: '#/components/schemas/Customer'
+        customerPaymentServiceProviderData:
+          $ref: '#/components/schemas/CustomerPaymentServiceProviderData'
+      required:
+        - customerPaymentServiceProviderData
+
+    CustomerResponse:
+      properties:
+        customer:
+          $ref: '#/components/schemas/Customer'
+
+    PreOrderPaymentOrderData:
+      description: 'Order data with optional fields to be used for pre order payments.'
+      type: object
+      required:
+        - grandTotal
+        - paymentMethod
+      properties:
+        orderReference:
+          type: string
+          nullable: true
+          example: 'DE--123456'
+        storeName:
+          type: string
+          nullable: true
+          example: 'DE'
+        localeName:
+          type: string
+          nullable: true
+          example: en_US
+        countryCode:
+          type: string
+          nullable: true
+          example: 'DE'
+        customerFirstName:
+          type: string
+          nullable: true
+          example: 'John'
+        customerLastName:
+          type: string
+          nullable: true
+          example: 'McClane'
+        currencyCode:
+          type: string
+          nullable: true
+          example: 'EUR'
+        grandTotal:
+          type: integer
+          example: 10000
+          description: 'Total in lowest denominator (cents)'
+        paymentMethod:
+          type: string
+          example: 'Credit Card'
+        items:
+          type: array
+          items:
+            $ref: '#/components/schemas/QuoteItem'
+          minItems: 1
+        additionalPaymentData:
+          $ref: '#/components/schemas/PreOrderPaymentAdditionalPaymentData'
+
+    OrderData:
+      description: 'Order data.'
+      type: object
+      required:
+        - countryCode
+        - customerFirstName
+        - customerLastName
+        - currencyCode
+        - grandTotal
+        - paymentMethod
+        - items
+      properties:
+        orderReference:
+          type: string
+          nullable: true
+          example: 'DE--123456'
+        storeName:
+          type: string
+          nullable: true
+          example: 'DE'
+        localeName:
+          type: string
+          nullable: true
+          example: en_US
+        countryCode:
+          type: string
+          example: 'DE'
+        customerFirstName:
+          type: string
+          example: 'John'
+        customerLastName:
+          type: string
+          example: 'McClane'
+        currencyCode:
+          type: string
+          example: 'EUR'
+        grandTotal:
+          type: integer
+          example: 10000
+          description: 'Total in lowest denominator (cents)'
+        paymentMethod:
+          type: string
+          example: 'Credit Card'
+        items:
+          type: array
+          items:
+            $ref: '#/components/schemas/QuoteItem'
+          minItems: 1
```

Comment:
currently, Stripe does not require `items`

Reply:
It will be latest when we add partial operations.